src/server/engine_shard_set.cc

```diff
@@ -324,7 +326,7 @@ bool EngineShard::DoDefrag() {
       }
     });
     traverses_count++;
-  } while (traverses_count < kMaxTraverses && cur);
+  } while (traverses_count < kMaxTraverses && cur && namespaces.IsInitialized());
```

Comment:
I do not understand this change

Reply:
Defrag is a background task that can trigger (mostly in tests) before the namespaces is initialized or after it is destroyed